lib/filterx/expr-variable.c

```diff
@@ -84,7 +80,14 @@ _eval(FilterXExpr *s)
     }
 
   if (!filterx_variable_handle_is_floating(self->handle))
-    return _pull_variable_from_message(self, context, context->msgs[0]);
+    {
+      FilterXObject *msg_ref = _pull_variable_from_message(self, context, context->msgs[0]);
+      if(!msg_ref)
+        goto error;
```

Comment:
I think we should directly return with `NULL` here, to retain the eval error message set in `_pull_variable_from_message()`. Jumping to `error` rewrites it.